zebra-state/src/service/finalized_state.rs

```diff
@@ -321,6 +347,77 @@ impl FinalizedState {
         result
     }
 
+    #[cfg(feature = "elasticsearch")]
+    /// Store finalized blocks into an elasticsearch database.
+    ///
+    /// We use the elasticsearch bulk api to index multiple blocks at a time while we are
+    /// synchronizing the chain, when we get close to tip we index blocks one by one.
+    pub fn elasticsearch(
+        &mut self,
+        block: &Arc<block::Block>,
+        committed_tip_height: &Option<block::Height>,
+    ) {
+        if let Some(client) = self.elastic_db.clone() {
+            let block_time = block.header.time.timestamp();
+            let local_time = chrono::Utc::now().timestamp();
+
+            const AWAY_FROM_TIP_BULK_SIZE: usize = 800;
+            const CLOSE_TO_TIP_BULK_SIZE: usize = 2;
+            const CLOSE_TO_TIP_SECONDS: i64 = 14400; // 4 hours
+
+            // If we are close to the tip index one block per bulk call.
+            let mut blocks_size_to_dump = AWAY_FROM_TIP_BULK_SIZE;
+            if local_time - block_time < CLOSE_TO_TIP_SECONDS {
+                blocks_size_to_dump = CLOSE_TO_TIP_BULK_SIZE;
+            }
+
+            // Insert the operation line.
+            let height_number = committed_tip_height.unwrap_or(block::Height(0)).0;
+            self.elastic_blocks.push(
+                serde_json::json!({
+                    "index": {
+                        "_id": height_number.to_string().as_str()
+                    }
+                })
+                .to_string(),
+            );
+
+            // Insert the block itself.
+            self.elastic_blocks
+                .push(serde_json::json!(block).to_string());
```

Comment:
Is there any block data that should be excluded here?

Reply:
I am not sure, probably not everything is useful or makes sense but for now i am just dumping the entire block into the database, users can filter stuff they don't want to see from the front end.